### DISCUSSION TITLE
No data after updating to 1.7.2

### DISCUSSION THREAD

**Discussion by mal5305** [jertel/vuegraf#159] [Q&A]
Hello, I'm currently running Vuegraf via Docker. Today I updated to 1.7.2 and am no longer getting any data to Influx. I've confirmed that live data is in my Emporia account via the app. Looking at Docker logs I see this repeating:
```
Attempt 1 to get device usage
Bad data for device 204305 channel 1,2,3 at 2024-02-28 19:59:23+00:00
Waiting 2 seconds before retrying device usage request.

Attempt 2 to get device usage
Bad data for device 204305 channel 1,2,3 at 2024-02-28 19:59:23+00:00
Waiting 4 seconds before retrying device usage request.

Attempt 3 to get device usage
Bad data for device 204305 channel 1,2,3 at 2024-02-28 19:59:23+00:00
2024-02-28 19:59:40.276100+00:00 | INFO  | Submitting datapoints to database; account="Home"; points=0
```

What information can I provide or look at to help troubleshoot? Thanks!

**Comment by mal5305** [comment]
I updated *from* 1.7.1

**Comment by jertel** [comment]
The new version uses a library that was changed to detect bad data coming back from Emporia. So you're seeing the problem now because the previous versions ignored the bad data. Specifically, it appears that your `1,2,3` channel (which I think represents the mains/high amperage sensor) isn't returning any data. Do you see data for that channel in the app?

**Reply by mal5305** [reply]
hey thanks for the reply. yes, there is data for that channel in the app, as well as all the other channels/circuits.

**Comment by mal5305** [comment]
@jertel i think i figured it out. i had an old smart plug that wasn't in use anymore on my account. i removed it and restarted the container, now requests are working as expected.

i really wish i'd have tested the API to see the response before i removed it from the app, though.

sounds like something unexpected comes through when a non-functioning plug is still added in the app. if you'd like me to test/reproduce to potentially build a fix, let me know. i'm happy to help.

**Reply by jertel** [reply]
That's great, I'm glad you figured it out. This is more of a problem with the [pyemvue](https://github.com/magico13/PyEmVue) library that I use in Vuegraf. It handles the comms to Emporia. The change made a few days ago is here:

https://github.com/magico13/PyEmVue/pull/65/files#diff-ff8845abff277daeda345072de6717780dafd6bd5dc7ee0e0dcf724b42c4c48f

It's abandoning the device collection loop if it encounters a device channel that has no data. I suggest creating an issue on that project and see if the author would be open to modifying it to allow it to continue processing sibling devices even if one device has no data. Perhaps an optional `ignore_bad_data=True` argument could be provided.
